webapp/src/features/intersections/reports/graphs/bar-chart-component.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, TooltipProps } from 'recharts'
+import { Box, Typography } from '@mui/material'
+import { ValueType, NameType } from 'recharts/types/component/DefaultTooltipContent'
+import { formatAxisTickNumber } from '../report-utils'
+
+interface BarChartComponentProps {
+  title: string
+  data: { name: string; value: number }[]
+  getInterval: (dataLength: number) => number
+  barColor: string
+}
+
+const CustomTooltip = ({ active, payload, label }: TooltipProps<ValueType, NameType>) => {
+  if (active && payload && payload.length) {
```

Comment:
suggestion(non-blocking): use optional chaining to reduce verbosity and improve readability
```suggestion
  if (active && payload?.length) {
```

Reply:
updated